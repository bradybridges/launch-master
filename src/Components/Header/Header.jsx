import React from 'react';
import { Switch } from '@material-ui/core';
import { makeStyles, Typography, AppBar, Toolbar } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  toolbar: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: '0 20px',
  },
}));

export default function Header({ darkMode, toggleDarkMode }) {
  const classes = useStyles();
  return (
    <header>
      <AppBar position="static">
        <Toolbar className={classes.toolbar}>
          <Typography variant="h1">Launch Master</Typography>
          <div>
            <Typography style={{ display: 'inline'}}>Light</Typography>
            <Switch
              checked={darkMode}
              onChange={() => toggleDarkMode()}
              color="secondary"
              name="darkmode-toggle"
              inputProps={{ 'aria-label': 'dark mode checkbox' }}
            />
            <Typography style={{ display: 'inline' }} color="secondary">Dark</Typography>
          </div>
        </Toolbar>
      </AppBar>
    </header>
  );
}
