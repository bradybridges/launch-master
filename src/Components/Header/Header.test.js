import React from 'react';
import { Switch, Typography } from '@material-ui/core';
import { shallow } from 'enzyme';
import Header from './Header';

describe('Header', () => {
  const mockToggleDarkMode = jest.fn();
  const wrapper = shallow(<Header  toggleDarkMode={mockToggleDarkMode} darkMode={true}/>);
  it('Header should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('There should be one switch button present in the header to toggle dark and light theme', () => {
    expect(wrapper.find(Switch)).toHaveLength(1);
  });

  it('When the switch is clicked toggleDarkMode should be called', () => {
    const toggle = wrapper.find(Switch);
    toggle.props().onChange();
    expect(mockToggleDarkMode).toHaveBeenCalled();
  });

  it('The dark mode toggle should be accompanied by typographys with children light and dark', () => {
    const light = wrapper.find(Typography).at(1).props();
    const dark = wrapper.find(Typography).at(2).props();
    const expectedLight = { style: { display: 'inline' }, children: 'Light' };
    const expectedDark = { style: { display: 'inline' }, color: 'secondary', children: 'Dark' };
    expect(light).toEqual(expectedLight);
    expect(dark).toEqual(expectedDark);
  })

  it('There should be an h1 header with a value of Launch Master', () => {
    const header = wrapper.find(Typography).first().props();
    expect(header.variant).toEqual('h1');
    expect(header.children).toEqual('Launch Master');
  })

})