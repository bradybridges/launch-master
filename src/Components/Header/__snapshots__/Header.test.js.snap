// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header should match snapshot 1`] = `
<header
  style={
    Object {
      "flexGrow": 1,
    }
  }
>
  <WithStyles(ForwardRef(AppBar))>
    <WithStyles(ForwardRef(Toolbar))
      className="makeStyles-toolbar-1"
    >
      <WithStyles(ForwardRef(Typography))
        variant="h1"
      >
        Launch Master
      </WithStyles(ForwardRef(Typography))>
      <div>
        <ForwardRef(SvgSun)
          className="makeStyles-darkModeIcon-5 makeStyles-rotate-6"
          onClick={[MockFunction]}
          style={
            Object {
              "fill": undefined,
            }
          }
        />
      </div>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</header>
`;
